cabby/rvs/templates.py

```diff
@@ -301,5 +309,21 @@ def add_features_to_template(template: Text, entity: geo_item.GeoEntity) -> Text
   template = template.replace(' a o', ' an o')
   template = template.replace('_', ' ')
 
-  return template
+  entities_span_list = {}
+  for entity_tag in entities_tags:
+    add_entity_span(entities_span_list, entity_tag, template)
+
+  return template, entities_span_list
+
+def add_entity_span(entities_span_list: Dict, entity_tag: str, instruction: str):
```

Comment:
This function has a side-effect of adding an item to the dictionary. It would be better to instead get back a dictionary from this function and then say entity_span_list.update(get_entity_spans(entity_tag, instruction)).

Reply:
Changed to entities_span_dict.update(add_entity_span(entity_tag, template))